src/app/mork/models/users.py

```diff
@@ -0,0 +1,94 @@
+"""Mork users models."""
+
+import enum
+from typing import List
+from uuid import uuid4
+
+from sqlalchemy import (
+    Enum,
+    ForeignKey,
+    Index,
+    Integer,
+    String,
+    UniqueConstraint,
+)
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from mork.models import Base
+
+
+class ServiceName(str, enum.Enum):
+    """Enum for service names."""
+
+    ASHLEY = "ashley"
+    EDX = "edx"
+    BREVO = "brevo"
+    JOANIE = "joanie"
+
+
+class DeletionReason(str, enum.Enum):
+    """Enum for the reason of the deletion request."""
+
+    USER_REQUESTED = "user_requested"
+    GDPR = "gdpr"
+
+
+class DeletionStatus(str, enum.Enum):
+    """Enum for deletion statuses."""
+
+    TO_DELETE = "to_delete"
+    DELETED = "deleted"
```

Comment:
IMO a `deleting` status should be useful.

We should have `to_delete` -> `deleting` -> `deleted`. Doing this, we can prevent race condition by preventing to process twice the same user

Reply:
You're right!